src/main/java/com/ludo/study/studymatchingplatform/common/utils/LocalDateTimePickerImpl.java

```diff
@@ -0,0 +1,15 @@
+package com.ludo.study.studymatchingplatform.common.utils;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+
+@Profile("!test")
+@Component
+public class LocalDateTimePickerImpl implements LocalDateTimePicker {
+
+    public LocalDateTime now() {
```

Comment:
```suggestion
    public LocalDateTime now() {
        return LocalDateTime.now().truncatedTo(ChronoUnit.MICROS);
    }
```

젠킨스 CI/CD 구축 과정에서 `Ubuntu` 환경에서 `LocalDateTime.now() / JpaAuditing`을 사용 시 `나노초(ns)까지 계산`되는 이슈를 확인했습니다.
`mac-os) LocalDateTime.now()` → 2024-05-22T13:37:23.054316
`ubuntu) LocalDateTime.now()` → 2024-05-22T13:37:23.054316123

현재 `LocalDateTimePickerImpl.now()` 메서드의 경우, 리뷰 가능 시간 계산을 위해서만 사용되고 있기 때문에 별다른 이슈가 발생하진 않을 것 같습니다. (참고) `LocalDateTimePickerImpl.now()`값이 DB에 저장되면 예상친 이슈가 발생할 수도 있습니다. timestamp(6) 컬럼이 소수점 6자리까지만 지원하여, ns초는 반올림처리하기 때문입니다.

정리하면, 현재 이슈가 될 것 같진 않지만 혹시 모를 상황을 대비하여 위의 suggest를 제안드려봅니다.

Reply:
문제 1. 현재 저희 비즈니스에 날짜와 관련한 요구사항이 늘어가고 있습니다.
ex1) 리뷰 가능 시간
ex2) 스터디 주간 참여 요일

문제 2. 그리고 위 코멘트에서 언급드린대로, `LocalDateTime.now()`에 대한 이슈 발생 가능성이 존재하고, `LocalDateTime.now().truncatedTo(ChronoUnit.MICROS)` 를 제안드렸습니다. 현재 저희 도메인 엔티티의 날짜 관련 필드는 LocalDateTime을 사용하고 있고, 이에 따라 중복 코드가 발생할 수도 있을 것 같습니다.


도메인 엔티티 필드의 `LocalDateTime` 타입들을 `LocalDateTimePicker`로 변경하는 것은 어떤지 여쭤보고 싶습니다.
빽님도 아시는 것처럼, 아래와 같은 장점이 있을 것으로 생각됩니다.
1. 제어할 수 있는 것에 의존
2. 날짜 관련 비즈니스 로직을 응집할 수 있음 (문제 1 개선 가능)
3. 날짜 관련 중복 코드를 재사용할 수 있음 (문제 2 개선 가능)


---
확인해야 할 기술적 이슈는 두 가지 정도 있을 것 같습니다.
1. LocalDateTime 래핑 클래스를 JPA에 적용 가능 여부
2. JpaAuditing 을 custom하여 사용하고 있는데, 이 또한 적용 가능 여부 (common/auditing/LocalDateTimeProvider.class 참고)

Reply:
`LocalDateTime.now()`로 값이 할당되는 로직을 `LocalDateTimePicker.now()`로 전부 바꾸자는 말씀이시죠??
(Custom JpaAudit 작성하여 `@CreatedDate`, `@LastModifiedDate`에 적용하는 부분 포함)

제가 이해한게 맞다면 해당 방법을 적용하는 것이 유지 보수에도 좋을 것 같습니다.

말씀하신대로 이전에 겪으신 테스트 시의 문제를 한 번의 변경으로 해결할 수 있을 것이고,

이전에 프론트엔드 분들이 별도의 Timezone 정보는 저장하지 않고, UTC+00 그리니치 시간으로 시간을 저장하는 식으로 이야기를 하셨던 것이 기억이 나는데, `LocalDateTime.now()`를 사용하면 UTC+09 한국 시간으로 적용되기 때문에 이 부분도 한 번의 변경으로 해결될 것 같네요.👍

Reply:
마지막 부분 말을 이상하게 해서 다시 정리..

LocalDateTime.now()를 사용하면 UTC+09 한국 시간으로 적용되기 때문에 현재는 **UTC+00 그리니치 시간으로 시간을 저장**하는 요구사항을 충족하지 못하고 있는데, 이 부분도 `LocalDateTimePicker`에서 한 번만 변경하면 해결될 것 같습니다.👍

Reply:
넵 맞습니다. 조금 정리해서 말씀드리면 아래와 같습니다!

1. interface LocalDateTimePicker
```java
public interface LocalDateTimePicker {
    LocalDateTime now();
}
```

2. class LocalDateTimePickerImpl
```java
public class LocalDateTimePickerImpl implements LocalDateTimePicker {
    public LocalDateTime now() {
        return LocalDateTime.now().truncatedTo(ChronoUnit.MICROS);
    }
}
```

3. 기존의 `LocalDateTime` 필드들을 모두 `LocalDateTimePicker` 로 대체
```java
//ex
// 애노테이션 등 생략
public abstract class BaseEntity {
    private LocalDateTimePicker createdDateTime;
    private LocalDateTimePicker updatedDateTime;
    private LocalDateTimePicker deletedDateTime = null;
}

public class Recruitment extends BaseEntity {
    // ...
    private LocalDateTimePicker recruitmentEndDateTime;
    // ...
}

// 다른 엔티티들도 동일하게 쭉쭉
```

4. Custom JpaAuditing 로직도 `LocalDateTimePicker` 로 수정
```java
@Component
public class LocalDateTimeProvider implements DateTimeProvider {
	@Override
	public Optional<TemporalAccessor> getNow() {
		return Optional.of(LocalDateTimePicker.now());
	}
}
```

5. LocalDateTime 직접 의존하여 사용하고 있는 로직들이 있다면, LocalDateTimePicker로 대체

----
@zxcev 저희가 이해하고 있는 내용이 일치하다면, 위의 내용들을 해당 PR에 반영해주실 수 있을까요? 🙂
아니면 다른 브랜치도 괜찮습니다!